cli/githubutil.py

```diff
@@ -140,3 +141,101 @@ def release_note_blocks_cli(
         repository_branch=repository_branch,
         commit_range=commit_range
     ).release_note_blocks()
+
+
+def cleanup_draft_releases(
+    github_cfg_name: str,
+    github_repository_owner: str,
+    github_repository_name: str,
+    latest_release_version: str,
+    dry_run: bool=True,
+):
+    github_cfg = ctx().cfg_factory().github(github_cfg_name)
+
+    githubrepobranch = GitHubRepoBranch(
+        github_config=github_cfg,
+        repo_owner=github_repository_owner,
+        repo_name=github_repository_name,
+        branch='master',
+    )
+
+    helper = GitHubRepositoryHelper.from_githubrepobranch(
+        githubrepobranch=githubrepobranch,
+    )
+    # prepare semver for later comparison
+    latest_semver = semver.parse_version_info(latest_release_version)
+
+    # we're only interested in releases whose name can be parsed using semver and whose semver
+    # contains the prerelease-label 'draft'.
+    def _is_semver_parseable(release):
+        try:
+            release_semver = semver.parse_version_info(release.name)
+        except Exception:
+            return False
+        if release_semver.prerelease != 'draft':
+            return False
+        return True
+
+    releases = helper.repository.releases()
+    # filter out non-draft releases
+    draft_releases = filter(lambda r: r.draft, releases)
+    # filter out draft-releases whose names aren't parseable
+    semver_releases = filter(_is_semver_parseable, draft_releases)
+
+    # returns a dict containing the release and the version_info object obtained
+    # from parsing it using semver.
+    def _to_release_and_semver(release):
+        return {
+            'release': release,
+            'version info': semver.parse_version_info(release.name),
+            }
+
+    # convert every release to dict containing release and semver (to avoid parsing the version
+    # repeatedly). This will be needed more than once, so persist it
+    draft_releases_and_semver = list(map(_to_release_and_semver, semver_releases))
```

Comment:
use a list comprehension if you need a list anyway. also, if using functional approach, construct processing chains with fluentiterable. otherwise, this just makes things harder to read by adding more boilerplate